cfdm/data/data.py

```diff
@@ -823,10 +866,13 @@ def _set_Array(self, array, copy=True):
 
         """
         if not isinstance(array, abstract.Array):
-            if not isinstance(array, numpy.ndarray):
-                array = numpy.asanyarray(array)
-
-            array = NumpyArray(array)
+            if not isinstance(array, np.ndarray):
+                if issparse(array):
```

Comment:
I see, but we don't need this because it's nested inside an `if not isinstance(array, abstract.Array):`, so if we have a `cf.SparseArray` we're not going to get into this block of code.

Reply:
Aha, gotcha. Good point. Marking as resolved.